catalog/IndexScheme.hpp

```diff
@@ -150,27 +146,22 @@ class IndexScheme {
    * @note This method assumes that the caller has already acquired the
    *       necessary locks before invoking it.
    *
-   * @param index_name The name of index to add (key)
-   * @param index_description The index description for this index (value)
+   * @param index_name The name of index to add (key).
+   * @param index_description The index description for this index (value).
    **/
   bool addIndexMapEntry(const std::string &index_name,
-                        const std::vector<IndexSubBlockDescription> &index_descriptions) {
+                        const IndexSubBlockDescription &index_description) {
```

Comment:
Since you `move index_description` later, it should not be marked as `const`. Instead, change to `IndexSubBlockDescription &&index_description` and the caller to use `move`.